lib/Dialect/FIRRTL/Transforms/SimplifyAggregates.cpp

```diff
@@ -244,12 +247,44 @@ bool MergeConnection::peelConnect(StrictConnectOp connect) {
   return true;
 }
 
-bool MergeConnection::run() {
+// Return true if multibit mux was folded into subaccess op.
+// If multbit mux inputs are `a[n-1], a[n-2], ..., a[1], a[0]`, we can fold
+// multibit into subaccess op.
+bool SimplifyAggregate::simplifyMultibitMux(MultibitMuxOp multibitMux) {
+
+  // Sample the last element of multibit mux inputs.
+  auto subindex = multibitMux.getInputs().back().getDefiningOp<SubindexOp>();
+  if (subindex.getIndex() != 0)
```

Comment:
I don't know that this short circuit really does much to save time since the loop terminates early.

Reply:
I was missing a condition to check `subindex.getInput() == sub.getInput()`. That's why a last element is extracted first.